File: OOP Start/PhonebookApp/UI/ConsoleView.java
package PhonebookApp.UI;

import java.time.LocalDate;
import java.util.Scanner;

import PhonebookApp.Config;
import PhonebookApp.Core.MVP.View;

public class ConsoleView implements View {
    Scanner in;
    public ConsoleView() {
        in = new Scanner(System.in, "Cp866");
    }

    @Override
    public LocalDate getDate() {
        System.out.printf("Date in format [%s]: ", Config.dateFormat);
        return LocalDate.parse(in.nextLine(), Config.ofPattern());
    }

    @Override
    public void setDate(LocalDate value) {
        System.out.printf("Date: %s\n", value.format(Config.ofPattern()));
    }

    @Override
    public String getFirstName() {
        System.out.printf("FirstName: ");
        return in.nextLine();
    }

    @Override
    public void setFirstName(String value) {
        System.out.printf("FirstName: %s\n", value);
    }

    @Override
    public String getLastName() {
        System.out.printf("LastName: ");
        return in.nextLine();
    }

    @Override
    public void setLastName(String value) {
        System.out.printf("LastName: %s\n", value);
    }

    @Override
    public String getPhone() {
        System.out.printf("Phone: ");
        return in.nextLine();
    }

    @Override
    public void setPhone(String value) {
        System.out.printf("Phone: %s\n", value);
    }

    @Override
    public String getCompany() {
        System.out.printf("Company: ");
        return in.nextLine();
    }

    @Override
    public void setCompany(String value) {
        System.out.printf("Company: %s\n", value);
    }

    @Override
    public String getEmail() {
        System.out.printf("Email: ");
        return in.nextLine();
    }

    @Override
    public void setEmail(String value) {
        System.out.printf("Email: %s\n", value);
    }

    @Override
    public String getDescription() {
        System.out.printf("Description: ");
        return in.nextLine();
    }

    @Override
    public void setDescription(String value) {
        System.out.printf("Description: %s\n", value);
        System.out.println();      
    }
    
    @Override
    public String[] getImportInfo() {
        System.out.println("0 return, 1 Import from CSV, 2 Import from TXT");
        String key = in.next();
        while (true) {
            switch (key) {
                case "0":
                    return null;
                case "1":
                    return new String[] {"csv", in.next()};
                case "2":
                    return new String[] {"txt", in.next()};
                default:
                    System.out.println("No such command");
                    break;
            }
        }
    }

    @Override
    public String[] getExportInfo() {
        System.out.println("0 return, 1 Export to CSV, 2 Export to TXT");
        String key = in.next();
        while (true) {
            switch (key) {
                case "0":
                    return null;
                case "1":
                    return new String[] {"csv", in.next()};
                case "2":
                    return new String[] {"txt", in.next()};
                default:
                    System.out.println("No such command");
                    break;
            }
        }
    }

    @Override
    public String[] getSearchInfo() {
        System.out.print("Enter query string: ");
        String key = in.next();
        return new String[] { key };
    }
}

File: OOP Start/OOP. Lesson 7/UI/ConsoleView.py
import datetime

from Core.MVP.View import View

class ConsoleView(View):
    def __init__(self):
        self.inp = input

    def getDate(self):
        dateFormat = "%Y-%m-%d"
        date_str = self.inp("Date in format [{}]: ".format(dateFormat))
        return datetime.datetime.strptime(date_str, dateFormat).date()

    def setDate(self, value):
        print("Date: {}".format(value.strftime("%Y-%m-%d")))

    def getFirstName(self):
        return self.inp("FirstName: ")

    def setFirstName(self, value):
        print("FirstName: {}".format(value))

    def getLastName(self):
        return self.inp("LastName: ")

    def setLastName(self, value):
        print("LastName: {}".format(value))

    def getPhone(self):
        return self.inp("Phone: ")

    def setPhone(self, value):
        print("Phone: {}".format(value))

    def getCompany(self):
        return self.inp("Company: ")

    def setCompany(self, value):
        print("Company: {}".format(value))

    def getEmail(self):
        return self.inp("Email: ")

    def setEmail(self, value):
        print("Email: {}".format(value))

    def getDescription(self):
        return self.inp("Description: ")

    def setDescription(self, value):
        print("Description: {}".format(value))
        print()

    def getImportInfo(self):
        print("0 return, 1 Import from CSV, 2 Import from TXT")
        key = self.inp()
        while True:
            if key == "0":
                return None
            elif key == "1":
                return ["csv", self.inp()]
            elif key == "2":
                return ["txt", self.inp()]
            else:
                print("No such command")
                key = self.inp()

    def getExportInfo(self):
        print("0 return, 1 Export to CSV, 2 Export to TXT")
        key = self.inp()
        while True:
            if key == "0":
                return None
            elif key == "1":
                return ["csv", self.inp()]
            elif key == "2":
                return ["txt", self.inp()]
            else:
                print("No such command")
                key = self.inp()

    def getSearchInfo(self):
        key = self.inp("Enter query string: ")
        return [key]
